prog/main.go

```diff
@@ -138,6 +142,41 @@ type appFlags struct {
 	consulInf       string
 }
 
+type containerLabelFiltersFlag struct {
+	apiTopologyOptions []app.APITopologyOption
+	exclude            bool
+}
+
+func (c *containerLabelFiltersFlag) String() string {
+	return fmt.Sprint(c.apiTopologyOptions)
+}
+
+func (c *containerLabelFiltersFlag) Set(flagValue string) error {
+	newAPITopologyOption, err := c.toAPITopologyOption(flagValue)
+	c.apiTopologyOptions = append(c.apiTopologyOptions, newAPITopologyOption)
+	return err
+}
+
+func (c *containerLabelFiltersFlag) toAPITopologyOption(flagValue string) (app.APITopologyOption, error) {
+	var indices []int
+	if indices = colonFinder.FindStringIndex(flagValue); len(indices) == 0 {
+		return app.APITopologyOption{}, fmt.Errorf("No unescaped colon found. This is needed to separate the title from the label")
+	} else if len(colonFinder.FindAllStringIndex(flagValue, -1)) > 1 {
+		return app.APITopologyOption{}, fmt.Errorf("Multiple unescaped colons. Escape colons that are part of the title and label")
+	}
+	var labelKeyValuePair []string
+	containerFilterTitle, containerFilterLabel := unescapeBackslashes.ReplaceAllString(flagValue[0:indices[0]+1], `$1`), unescapeBackslashes.ReplaceAllString(flagValue[indices[1]:], `$1`)
+	if labelKeyValuePair = strings.Split(containerFilterLabel, "="); len(labelKeyValuePair) != 2 {
+		return app.APITopologyOption{}, fmt.Errorf("Docker label in the app.container-label-filter(-exclude) flag isn't in the correct key=value format")
+	}
+	switch {
+	case c.exclude:
+		return app.MakeAPITopologyOption(fmt.Sprintf("containerLabelFilterExclude%d", len(c.apiTopologyOptions)), containerFilterTitle, render.DoesNotHaveLabel(labelKeyValuePair[0], labelKeyValuePair[1]), false), nil
```

Comment:
I am not sure what you mean (since the ID is dynamic). But give it a go and see if it's easier to read.